third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -139,8 +145,19 @@ pub enum Type {
     Address,
     Signer,
     Vector(Box<Type>),
-    Struct(CachedStructIndex),
-    StructInstantiation(CachedStructIndex, Vec<Type>),
+    Struct {
+        name: Arc<StructName>,
+        #[derivative(PartialEq = "ignore", Hash = "ignore")]
+        ability: AbilitySet,
+    },
+    StructInstantiation {
+        name: Arc<StructName>,
+        ty_args: Vec<Type>,
+        #[derivative(PartialEq = "ignore", Hash = "ignore")]
+        base_ability_set: AbilitySet,
+        #[derivative(PartialEq = "ignore", Hash = "ignore")]
+        phantom_ty_args_mask: Vec<bool>,
```

Comment:
Nice trick! Do you also need to tell it to ignore these fields for Ord?

Reply:
That's why I changed the BTreeMap to HashMap! So it seems that this derivative crate can't handle Ord very well.. https://mcarton.github.io/rust-derivative/latest/cmp.html

Reply:
Then shall we also remove `#[derive(Ord, PartialOrd)]`?

Reply:
True. Lemme take a look.

Reply:
BTW the way I implement the struct type here might still make us vulnerable to long identifier attack. I'll use Arc instead of cloning to avoid memory consumption.